src/core/stream/orchestrator/stream_orchestrator.ts

```diff
@@ -130,10 +130,17 @@ export default function StreamOrchestrator(
                                                    (val) =>
                                                      Math.min(val, defaultMaxBehind),
                                                    gcCancelSignal),
-            maxBufferAhead: createMappedReference(maxBufferAhead,
-                                                  (val) =>
-                                                    Math.min(val, defaultMaxAhead),
-                                                  gcCancelSignal) },
+            maxBufferAhead: createMappedReference(maxBufferAhead, (val) => {
+              const actualMaxBuff = bufferType === "text" ?
+                // Text segments are both much lighter on resources and might
+                // actually be much larger than other types of segments in terms
+                // of duration. Let's make an exception here by authorizing a
+                // larger text buffer ahead, to avoid unnecesarily reloading the
+                // same text track.
+                Math.max(val, 2 * 60) :
```

Comment:
You can set this in the config file something like: `MINIMUM_MAX_BUFFER_AHEAD`

Reply:
Done